src/frozen_fields/main.py

```diff
@@ -36,114 +39,42 @@
-        sticky = [fld["sticky"] for fld in flds]
-        self.web.eval(js_code_20)
-        self.web.eval("setFrozenFields(%s, %s, %d);" % (
-            json.dumps(data), json.dumps(sticky), field))
-    else:
-        self.web.eval("setFields(%s, %d);" % (
-            json.dumps(data), field))
-    ###########################
-    self.web.eval("setFonts(%s);" % (
-        json.dumps(self.fonts())))
-    self.checkValid()
-    self.widget.show()
-    if self.stealFocus:
-        self.web.setFocus()
-        # self.stealFocus = False
-
-
-def loadNote21(self, focusTo=None):
-    if not self.note:
-        return
-
-    data = []
-    for fld, val in list(self.note.items()):
-        data.append((fld, self.mw.col.media.escapeImages(val)))
-    self.widget.show()
-    self.updateTags()
-
-    def oncallback(arg):
-        if not self.note:
-            return
-        self.setupForegroundButton()
-        self.checkValid()
-        if focusTo is not None:
-            self.web.setFocus()
-        try:
-            from aqt import gui_hooks
-            gui_hooks.editor_did_load_note(self)
-        except:
-            runHook("loadNote", self)
-
-    # only modify AddCards Editor
     if not isinstance(self.parentWindow, AddCards):
-        self.web.evalWithCallback("setFields(%s); setFonts(%s); focusField(%s); setNoteId(%s)" % (
-            json.dumps(data),
-            json.dumps(self.fonts()), json.dumps(focusTo),
-            json.dumps(self.note.id)),
-            oncallback)
-    else:
-        iconstr_frozen = self.resourceToData(icon_path_frozen)
-        iconstr_unfrozen = self.resourceToData(icon_path_unfrozen)
-
-        flds = self.note.model()["flds"]
-        sticky = [fld["sticky"] for fld in flds]
-
-        eval_definitions = js_code_21 % (hotkey_toggle_field, iconstr_frozen,
-                                         iconstr_unfrozen)
-
-        eval_calls = "setFrozenFields(%s, %s); setFonts(%s); focusField(%s); setNoteId(%s)" % (
-            json.dumps(data), json.dumps(sticky),
-            json.dumps(self.fonts()),
-            json.dumps(focusTo),
-            json.dumps(self.note.id))
+        return
+    iconstr_frozen = self.resourceToData(icon_path_frozen)
+    iconstr_unfrozen = self.resourceToData(icon_path_unfrozen)
+    flds = self.note.model()["flds"]
+    sticky = [fld["sticky"] for fld in flds]
 
-        self.web.eval(eval_definitions)
-        self.web.evalWithCallback(eval_calls, oncallback)
+    self.web.eval(f"""var hotkey_toggle_field = "{hotkey_toggle_field}";""")
+    self.web.eval(f"""var src_unfrozen = "{iconstr_unfrozen}";""")
+    self.web.eval(f"""var src_frozen = "{iconstr_frozen}";""")
+    self.web.eval(f"setFrozenFields({json.dumps(sticky)});")
 
 
-def onBridge(self, str, _old):
+def onBridge(handled, str, editor):
     """Extends the js<->py bridge with our pycmd handler"""
 
+    if not isinstance(editor, Editor) or not isinstance(editor.parentWindow, AddCards):
+        return
```

Comment:
This needs to return `handled`, otherwise entering any webview other than the editor causes:

```
TypeError: cannot unpack non-iterable NoneType object
```